pubsub/gcp/gcp.go

```diff
@@ -133,15 +130,26 @@ var _ pubsub.Publisher = &publisher{}
 var _ pubsub.MultiPublisher = &publisher{}
 
 // NewPublisher will instantiate a new GCP MultiPublisher.
-func NewPublisher(ctx context.Context, projID, topic string, opts ...option.ClientOption) (pubsub.MultiPublisher, error) {
-	client, err := gpubsub.NewClient(ctx, projID, opts...)
+func NewPublisher(ctx context.Context, cfg Config) (pubsub.MultiPublisher, error) {
```

Comment:
Is there any reason to remove the `ClientOption`s being passed to the construct?